### PR TITLE
bye travis! replaces travis with github actions

### PR DESCRIPTION
**Pull request by LetterN**
title.

oh yeah it also updates the tools for frictionless devving (mapmerge, 3way merge feature. midi2piano is py, built in tool to install python for cbt)
all maps rebaked into tgm because some aren't tgm
all the tools are updated to latest

🆑
add: CI
code: Tools updated
code: all maps are converted to tgm
code: forced LF on other files
code: dmdoc building
code: varset callbacks
code: qdel will NOT complain about deleting things that aren't datum (like the world)
code: updated callbacks
code: forced (any encoding) => utf8
/🆑

# if you are merging this please make sure it is using the squashed strategy

### CODE REVIEW

**Comment by drexample** [comment]
pls use the changelog thingie, I don't want it to make a sad x mark in action logs :C

**Comment by LetterN** [comment]
> pls use the changelog thingie, I don't want it to make a sad x mark in action logs :C

x mark in action logs? or changelog?

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+trim_trailing_whitespace = true
+insert_final_newline = true
+end_of_line = lf
+
+[*.yml]
+indent_style = space
+indent_size = 2
+
+[*.py]
+indent_style = space
```
Comment (AlexMorgan3817) [inline .editorconfig:14]:
```suggestion
indent_style = tab
```

File: .github/workflows/compile_changelogs.yml [hunk 2]
```diff
@@ -2,34 +2,51 @@ name: Compile changelogs
...
+          if [ -n $CHANGELOG_ENABLER ]; then SECRET_EXISTS='true' ; fi
+          echo ::set-output name=CL_ENABLED::${SECRET_EXISTS}
+      - name: "Setup python"
+        if: steps.value_holder.outputs.CL_ENABLED
         uses: actions/setup-python@v1
         with:
           python-version: '3.x'
-      - name: Install depends
+      - name: "Install deps"
+        if: steps.value_holder.outputs.CL_ENABLED
```
Comment (AlexMorgan3817) [inline .github/workflows/compile_changelogs.yml:26]: Oh, yeah, it's PR about Travis, i won't change anything in other hooks somewhy

File: .github/workflows/compile_changelogs.yml [hunk 3]
```diff
@@ -2,34 +2,51 @@ name: Compile changelogs
...
         run: |
           git config --local user.email "action@github.com"
-          git config --local user.name "Nestor Jr."
+          git config --local user.name "Changelogs"
           git pull origin master
+          git add html/changelogs
           git commit -m "Automatic changelog compile [ci skip]" -a || true
-      - name: Push
+      - name: "Push"
```
Comment (AlexMorgan3817) [inline .github/workflows/compile_changelogs.yml:48]: Why the heck you putting it in strings, YML already count it as string

File: .github/workflows/compile_changelogs.yml [hunk 4]
```diff
@@ -2,34 +2,51 @@ name: Compile changelogs
 
 on:
   schedule:
-    - cron: "* */12 * * *"
+  - cron: "0 0 * * *"
```
Comment (AlexMorgan3817) [inline .github/workflows/compile_changelogs.yml:5]: Why do you touching it?

File: .github/workflows/conflicts.yml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+    branches:
+      - master
+jobs:
+  triage:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: mschilde/auto-label-merge-conflicts@2e8fcc76c6430272ec8bb64fb74ec1592156aa6a
+        with:
+          CONFLICT_LABEL_NAME: 'Merge Conflict'
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (AlexMorgan3817) [inline .github/workflows/conflicts.yml:13]: Travis?

File: .github/workflows/generate_documentation.yml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+          touch dmdoc/.nojekyll
+          echo codedocs.tgstation13.org > dmdoc/CNAME
+      - name: Deploy
+        uses: JamesIves/github-pages-deploy-action@3.7.1
+        with:
+          BRANCH: gh-pages
+          CLEAN: true
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SINGLE_COMMIT: true
+          FOLDER: dmdoc
```
Comment (AlexMorgan3817) [inline .github/workflows/generate_documentation.yml:31]: Travis???

File: .gitignore [hunk 7]
```diff
@@ -1,22 +1,195 @@
...
+!/config/title_screens/images/exclude
+
+#Linux docker
+/tools/LinuxOneShot/SetupProgram/obj/*
+/tools/LinuxOneShot/SetupProgram/bin/*
+/tools/LinuxOneShot/SetupProgram/.vs
+/tools/LinuxOneShot/Database
+/tools/LinuxOneShot/TGS_Config
+/tools/LinuxOneShot/TGS_Instances
+/tools/LinuxOneShot/TGS_Logs
```
Comment (AlexMorgan3817) [inline .gitignore]: Okey, i will be short, you writed in PR only replacing Travis CI with github actions, and now, looking in, i see so many off-top changes, so you'd better to describe ALL of them, not just title.

File: cev_eris.dme [hunk 8]
```diff
@@ -1,2790 +1,2790 @@
...
-#include "code\game\machinery\ai_slipper.dm"
-#include "code\game\machinery\alarm.dm"
-#include "code\game\machinery\antigrav.dm"
-#include "code\game\machinery\atmo_control.dm"
-#include "code\game\machinery\autolathe_disk_cloner.dm"
-#include "code\game\machinery\Beacon.dm"
-#include "code\game\machinery\biogenerator.dm"
-#include "code\game\machinery\bioprinter.dm"
-#include "code\game\machinery\bioprinter_nt.dm"
-#include "code\game\machinery\bluespaceartillery.dm"
```
Comment (AlexMorgan3817) [inline cev_eris.dme:649]: Cringe

File: code/__DEFINES/unit_testing.dm [hunk 9]
```diff
@@ -1,10 +0,0 @@
-/*
- *
- *   This file is used by Travis to indicate that Unit Tests are to be ran.
- *   Do not add anything but the UNIT_TEST definition here as it will be overwritten by Travis when running tests.
- *
- *
- *   Should you wish to edit set UNIT_TEST to 1 like so:
- *   #define UNIT_TEST 1
- */
-#define UNIT_TEST 0
```
Comment (AlexMorgan3817) [inline code/__DEFINES/unit_testing.dm:10]: ??????

File: code/_compatibility/513.dm [hunk 10]
```diff
@@ -1,9 +1,3 @@
-#if DM_VERSION > 513
-#warn Remie said that lummox was adding a way to get a lists
-#warn contents via list.values, if that is true remove this
-#warn otherwise, update the version and bug lummox
-#endif
-
```
Comment (AlexMorgan3817) [inline code/_compatibility/513.dm:6]: ??????

File: code/_compile_options.dm [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+#endif
+
+#if defined(UNIT_TESTS)
+//Hard del testing defines
+#define REFERENCE_TRACKING
+#define REFERENCE_TRACKING_DEBUG
+#define FIND_REF_NO_CHECK_TICK
+#endif
+
+#ifdef TGS
```
Comment (AlexMorgan3817) [inline code/_compile_options.dm:85]: If you port fucking TG Suit, so mark it in PR, not just "Replacing Travis with Github actions"

File: code/global.dm [hunk 12]
```diff
@@ -1,8 +1,3 @@
-//#define TESTING
-#if DM_VERSION < 506
-#warn This compiler is out of date. You may experience issues with projectile animations.
-#endif
```
Comment (AlexMorgan3817) [inline code/global.dm:4]: WHY?

**Review by AlexMorgan3817 — CHANGES_REQUESTED** [review]
Author or inherit not master when tried to replace Travis and inherit his WIP TG Suit or something wrong with author that haven't describe all what he done here

**Comment by AlexMorgan3817** [comment]
Ёбик, короче

File: .editorconfig (hunk 1 above)
Comment (LetterN) [reply .editorconfig:14]: no. it's a python file

File: .github/workflows/compile_changelogs.yml (hunk 2 above)
Comment (LetterN) [reply .github/workflows/compile_changelogs.yml:26]: it's a secret var maintainer has to add on git repo, already done

File: .github/workflows/conflicts.yml (hunk 5 above)
Comment (LetterN) [reply .github/workflows/conflicts.yml:13]: gone

File: .github/workflows/generate_documentation.yml (hunk 6 above)
Comment (LetterN) [reply .github/workflows/generate_documentation.yml:31]: eliminated

File: .gitignore (hunk 7 above)
Comment (LetterN) [reply .gitignore]: it's just gitignore pulled from tg. This just blocks generic useless things added by the os/code editor

File: cev_eris.dme (hunk 8 above)
Comment (LetterN) [reply cev_eris.dme:649]: crlf -> lf

File: code/__DEFINES/unit_testing.dm (hunk 9 above)
Comment (LetterN) [reply code/__DEFINES/unit_testing.dm:10]: cbt injects it

File: code/_compatibility/513.dm (hunk 10 above)
Comment (LetterN) [reply code/_compatibility/513.dm:6]: it _can_ be removed

File: code/global.dm (hunk 12 above)
Comment (LetterN) [reply code/global.dm:4]: dupe check `compile_options.dm`

File: .editorconfig (hunk 1 above)
Comment (AlexMorgan3817) [reply .editorconfig:14]: Yes, it's python file and it should use tab, not shitty spaces

File: .github/workflows/compile_changelogs.yml (hunk 2 above)
Comment (AlexMorgan3817) [reply .github/workflows/compile_changelogs.yml:26]: It is still off-top

File: .github/workflows/generate_documentation.yml (hunk 6 above)
Comment (AlexMorgan3817) [reply .github/workflows/generate_documentation.yml:31]: No, i asked how it is related to Travis

File: .gitignore (hunk 7 above)
Comment (AlexMorgan3817) [reply .gitignore]: Why you do off-top actions in PR that described to remove Travis and replace it with actions

File: cev_eris.dme (hunk 8 above)
Comment (AlexMorgan3817) [reply cev_eris.dme]: Why we need to get from crlf to lf if this PR just replaces Travis with actions?

File: code/_compatibility/513.dm (hunk 10 above)
Comment (AlexMorgan3817) [reply code/_compatibility/513.dm:6]: It is still off-top, and this warns should be checked

**Comment by AlexMorgan3817** [comment]
PR still haven't description what author done with build and changelog of all off-top shit that add by this PR

**Comment by AlexMorgan3817** [comment]
> PR still haven't description what author done with build and changelog of all off-top that added by this PR

**Comment by drexample** [comment]
Please make the PR compliant to our PR template.

**Review by drexample — APPROVED** [review]

**Comment by TheShown911** [comment]
Is the CI integration supposed to fail?

**Comment by LetterN** [comment]
yes, due to missing sprite (borgs)

**Comment by TheShown911** [comment]
Well what could go wrong?

**Pull request merged**